src/widgets/RRule.tsx

```diff
@@ -0,0 +1,284 @@
+import * as React from 'react';
+import Container from './Container';
+import { TextField, Select, MenuItem, FormGroup, FormControlLabel, Checkbox, InputLabel, FormControl } from '@material-ui/core';
+import RRule, { Options, Weekday, Frequency, ByWeekday } from 'rrule';
+import { ALL_WEEKDAYS } from 'rrule/dist/esm/src/weekday';
+import DateTimePicker from '../widgets/DateTimePicker';
+
+interface PropsType {
+  onChange: (rrule: string) => void;
+  rrule: string;
+}
+enum Ends {
+  Never,
+  Date,
+  After,
+}
+enum Months {
+  Jan = 1,
+  Feb,
+  Mar,
+  Apr,
+  May,
+  Jun,
+  Jul,
+  Aug,
+  Sep,
+  Oct,
+  Nov,
+  Dec,
+}
+enum MonthReapet {
+  bysetpos,
+  bymonthday,
+}
+enum Position {
+  First = 1,
+  Second = 2,
+  Third = 3,
+  Fourth = 4,
+  Last = -1,
+}
+const menueItemsPosition = Object.keys(Position).filter((key) => Number(key)).map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{Position[key]}</MenuItem>
+  );
+});
+const menueItemsMonths = Object.keys(Months).filter((key) => Number(key)).map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{Months[key]}</MenuItem>
+  );
+});
+const menueItemsEnds = [Ends.Never, Ends.Date, Ends.After].map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{Ends[key]}</MenuItem>
+  );
+});
+const bysetposWeekDay = ALL_WEEKDAYS.map((value, index) => {
+  return (
+    <MenuItem key={index} value={index}>{value}</MenuItem>
+  );
+});
+const menuItemsFrequency = Object.keys(Frequency).filter((key) => Number(key) < 4).map((value) => {
+  return (
+    <MenuItem key={value} value={value}>{Frequency[value].toLowerCase()}</MenuItem>
+  );
+});
+const menuItemDaysNumbers = Array.from(Array(31).keys()).map((value: number) => {
```

Comment:
Having 31 menu items is ugly. Better make it a free field that people can just enter a number. Also, I think the numbers are not mutually exclusive, I don't think there should be just one choice, should there?

Reply:
I can change it to TextField with min 1 max 31

Reply:
bymonthday in rrule can be a list of numbers , what input would make sense for multi select in range  1 to 31?

Reply:
Textfield sounds good. As for multiselect in this range: just the ability to add more of these fields (so a + button and an x button, for removal), though I wouldn't worry about it in stage 1.

Have you looked how it's represented in other apps?

Reply:
![image](https://user-images.githubusercontent.com/42600279/71645481-dae62100-2ce1-11ea-9a9b-0877af79d2b3.png)
on google calendar they just don't support all the rrule options just the basic ones

Reply:
What about other apps?